### PR TITLE
Fix order of artists through mass edit

### LINKED ISSUE #34: Newly created artists aren't added in the right order on mass edit

### PR DESCRIPTION
**Pull request by robbevp**
## Description

Fix #34 
We previously set the order of the artists in a track based on the length, which changed while we were adding the artists.
This would mess up the order if for some reason the transformedArtists array is not in the same order as the order property. (Which seems to happen with new artists)

We now set the order based on the length of the length before we start adding.
I've dropped the `-1` since this would generate orders like `1, 1, 2, 3`.

# How Has This Been Tested?

Tested locally with a number of tracks

### CODE REVIEW

**Review by charvp — CHANGES_REQUESTED** [review]
This fix doesn't seem right to me. Changes to the `transformed` track shouldn't reflect on the original `t` track. Or am I missing something?

**Comment by robbevp** [comment]
> This fix doesn't seem right to me. Changes to the `transformed` track shouldn't reflect on the original `t` track. Or am I missing something?

You're right. Apparently the problem started when we create `transformed`: both `t.track_artists` and `transformed.track_artists` referred to the same array, so when we pushed to `transformed.track_artists` we also mutated `t.track_artists`. Fixed by destructuring and creating a new array

File: src/components/MassEditDialog.vue [hunk 1]
```diff
@@ -457,7 +457,7 @@ export default {
           album_id: t.album_id,
           review_comment: this.clearReviewComments ? null : t.review_comment,
           genre_ids: t.genre_ids,
```
Comment (charvp) [inline src/components/MassEditDialog.vue:459]: I guess this should also happen for `genre_ids`?

**Review by charvp — CHANGES_REQUESTED** [review]
Doesn't the -1 thing still apply as well?

File: src/components/MassEditDialog.vue (hunk 1 above)
Comment (robbevp) [reply src/components/MassEditDialog.vue]: true, fixed

**Review by charvp — APPROVED** [review]

**Pull request merged**